frontend/src/createStream/__tests__/StreamingTable.test.js

```diff
@@ -0,0 +1,239 @@
+    })
+
+
+    describe("handleSubmit", () => {
+        it("should do nothing if no receiver or sender has been selected", () => {
+
+            const data =  {
+                data: "test"
+            }
+            axios.post.mockImplementationOnce(()=>Promise.resolve(data))
+            // act
+            wrapper.instance().handleSubmit(new DummyData);
+            expect(axios.post).not.toHaveBeenCalled();
+        }),
+        it("should do nothing if a receiver but no sender has been selected", () => {
+            const mockReceiver = {
+                target: {
+                    name: "selectedReceiverID",
+                    value: "0_Test5_Test6"
+                }
+            };
+
+            const data =  {
+                data: "test"
+            }
+            axios.post.mockImplementationOnce(()=>Promise.resolve(data))
+
+            // act
+            wrapper.instance().onReceiverSelected(mockReceiver);
+            
+            wrapper.instance().handleSubmit(new DummyData);
+            expect(axios.post).not.toHaveBeenCalled();
+        }),
+        it("should do nothing if no receiver but a sender has been selected", () => {
+            const mockSender = {
+                target: {
+                    name: "selectedSenderID",
+                    value: "0_Test2_Test3"
+                }
+            };
+            const data =  {
+                data: "test"
+            }
+            axios.post.mockImplementationOnce(()=>Promise.resolve(data))
+
+
+            // act
+            wrapper.instance().onSenderSelected(mockSender);
+            wrapper.instance().handleSubmit(new DummyData);
+            expect(axios.post).not.toHaveBeenCalled();
+        }),
+
+        it("should call axios.post if a sender and a receiver have been selected", () => {
+
+            const mockReceiver = {
+                target: {
+                    name: "selectedReceiverID",
+                    value: "0_Test5_Test6"
+                }
+            };
+            const mockSender = {
+                target: {
+                    name: "selectedSenderID",
+                    value: "0_Test2_Test3"
+                }
+            };
+
+            const expected = {
+                inputChannelId: "Test3",
+                outputChannelId: "Test6"
+            };
+
+            const data =  {
+                data: "test"
+            }
+            axios.post.mockImplementationOnce(()=>Promise.resolve(data))
+
+            wrapper.instance().onSenderSelected(mockSender);            
+            wrapper.instance().onReceiverSelected(mockReceiver);
+
+            wrapper.instance().handleSubmit(new DummyData);
+
+            expect(axios.post).toHaveBeenCalledWith("http://localhost:8080/stream", expected);
+
+            setTimeout(
+                ()=>{
+                    expect(console.log).toHaveBeenCalledWith("Success. Stream Started.");
+                }
+                , 2000
+            )
```

Comment:
Because it takes a bit before it calls the success message. 
As discussed, this is because this puts it at the back of the queue, letting the console.log to get called. Will reduce the timeout to 0

Reply:
Fixed in aee4deb